tests/sql/operators/test_load_file.py

```diff
@@ -836,8 +836,43 @@ def test_aql_load_file_optimized_path_method_is_not_called(
 
     with mock.patch(mock_path) as method:
         load_file(
-            input_file=File(file_uri),
-            output_table=test_table,
-            use_native_support=False,
+            input_file=File(file_uri), output_table=test_table, use_native_support=False
         ).operator.execute({})
         assert not method.called
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.BIGQUERY,
+        }
+    ],
+    indirect=True,
+    ids=["Bigquery"],
+)
+def test_aql_load_file_optimized_path_method_is_not_called_1(
```

Comment:
@utkarsharma2 I'm slightly worried about the imbalance between the huge amount of code introduced by this PR and the little number of tests.

Should we try to cover some simple functions with unit test, where relevant, and perhaps handle some error scenarios?

Reply:
One option is to move the `kwargs_for_native_path` in a separate PR

Reply:
@kaxil the integration test won't without this change, as we need to pass certain parameters that are specific to this dataset.